packages/medusa/src/services/order-edit.ts

```diff
@@ -686,6 +628,116 @@ export default class OrderEditService extends TransactionBaseService {
+    orderEditId: string,
+    context: { loggedInUserId?: string } = {}
+  ): Promise<OrderEdit> {
+    return await this.atomicPhase_(async (manager) => {
+      const orderEditRepository = manager.getCustomRepository(
+        this.orderEditRepository_
+      )
+
+      let orderEdit = await this.retrieve(orderEditId)
+
+      if (
+        [OrderEditStatus.CANCELED, OrderEditStatus.DECLINED].includes(
+          orderEdit.status
+        )
+      ) {
+        throw new MedusaError(
+          MedusaError.Types.NOT_ALLOWED,
+          `Cannot confirm an order edit with status ${orderEdit.status}`
+        )
+      }
+
+      if (orderEdit.status === OrderEditStatus.CONFIRMED) {
+        return orderEdit
+      }
+
+      const lineItemServiceTx = this.lineItemService_.withTransaction(manager)
+
+      await Promise.all([
+        lineItemServiceTx.update(
+          { order_id: orderEdit.order_id },
+          { order_id: null }
+        ),
+        lineItemServiceTx.update(
+          { order_edit_id: orderEditId },
+          { order_id: orderEdit.order_id }
+        ),
+      ])
+
+      orderEdit.confirmed_at = new Date()
+      orderEdit.confirmed_by = context.loggedInUserId
+
+      orderEdit = await orderEditRepository.save(orderEdit)
+
+      await this.eventBusService_
+        .withTransaction(manager)
+        .emit(OrderEditService.Events.CONFIRMED, { id: orderEditId })
+
+      return orderEdit
+    })
+  }
+
+  protected async retrieveActive(
+    orderId: string,
+    config: FindConfig<OrderEdit> = {}
+  ): Promise<OrderEdit | undefined> {
+    const manager = this.transactionManager_ ?? this.manager_
+    const orderEditRepository = manager.getCustomRepository(
+      this.orderEditRepository_
+    )
+
+    const query = buildQuery(
+      {
+        order_id: orderId,
+        confirmed_at: IsNull(),
+        canceled_at: IsNull(),
+        declined_at: IsNull(),
+      },
+      config
+    )
+    return await orderEditRepository.findOne(query)
+  }
+
+  protected async deleteClonedItems(orderEditId: string): Promise<void> {
+    const manager = this.transactionManager_ ?? this.manager_
+    const lineItemServiceTx = this.lineItemService_.withTransaction(manager)
+    const lineItemAdjustmentServiceTx =
+      this.lineItemAdjustmentService_.withTransaction(manager)
+    const taxProviderServiceTs =
+      this.taxProviderService_.withTransaction(manager)
+
+    const clonedLineItems = await lineItemServiceTx.list(
+      {
+        order_edit_id: orderEditId,
+      },
+      {
+        select: ["id", "tax_lines", "adjustments"],
+        relations: ["tax_lines", "adjustments"],
+      }
+    )
+    const clonedItemIds = clonedLineItems.map((item) => item.id)
+
+    await Promise.all(
+      [
+        taxProviderServiceTs.clearLineItemsTaxLines(clonedItemIds),
+        clonedItemIds.map(async (id) => {
+          return await lineItemAdjustmentServiceTx.delete({
+            item_id: id,
+          })
+        }),
```

Comment:
question: is this necessary, since the line item is also deleted, cascade delete on the relation should take care of this

Reply:
After a discussion with @srindom we went that way to not rely on RDBMS specific features in case in the future we decide to split (let's say in multiple db or a different system). It is more "future proof"